cmd/kubeadm/app/util/etcd/etcd.go

```diff
@@ -360,8 +360,28 @@ func (c *Client) RemoveMember(id uint64) ([]Member, error) {
 			defer func() { _ = cli.Close() }()
 
 			ctx, cancel := context.WithTimeout(context.Background(), etcdTimeout)
+			defer cancel()
+
+			// list members and quick return if the member don't exist
+			listResp, err := cli.MemberList(ctx)
+			if err != nil {
+				klog.V(5).Infof("Failed to check whether the member %s exists, %v", strconv.FormatUint(id, 16), err)
+				lastError = err
+				return false, nil
+			}
+			found := false
+			for _, member := range listResp.Members {
+				if member.GetID() == id {
```

Comment:
i guess we need to rework our RemoveMember function if we want to match PeerURL here, but also not sure if we know the PeerURL when we are calling kubeadm reset?

Reply:
https://github.com/kubernetes/kubernetes/blob/4c24b9337fa189d99a721bfdc4785d62a744261b/cmd/kubeadm/app/phases/etcd/local.go#L116-L119